tensorboard/plugins/debugger/debugger_plugin_test.py

```diff
@@ -0,0 +1,436 @@
+# pylint: enable=ungrouped-imports, wrong-import-order
+
+
+class DebuggerPluginTest(tf.test.TestCase):
+
+  def setUp(self):
+    # Populate the log directory with debugger event for run '.'.
+    self.log_dir = tempfile.mkdtemp()
+    file_prefix = tf.compat.as_bytes(
+        os.path.join(self.log_dir, 'events.debugger'))
+    writer = tf.pywrap_tensorflow.EventsWriter(file_prefix)
+    device_name = '/job:localhost/replica:0/task:0/cpu:0'
+    writer.WriteEvent(
+        self._CreateEventWithDebugNumericSummary(
+            device_name=device_name,
+            op_name='layers/Matmul',
+            output_slot=0,
+            wall_time=42,
+            step=2,
+            list_of_values=(list(range(12)) +
+                            [float(tf.float32.as_datatype_enum), 1.0, 3.0])))
+    writer.WriteEvent(
+        self._CreateEventWithDebugNumericSummary(
+            device_name=device_name,
+            op_name='layers/Matmul',
+            output_slot=1,
+            wall_time=43,
+            step=7,
+            list_of_values=(
+                list(range(12)) +
+                [float(tf.float64.as_datatype_enum), 2.0, 3.0, 3.0])))
+    writer.WriteEvent(
+        self._CreateEventWithDebugNumericSummary(
+            device_name=device_name,
+            op_name='logits/Add',
+            output_slot=0,
+            wall_time=1337,
+            step=7,
+            list_of_values=(list(range(12)) +
+                            [float(tf.int32.as_datatype_enum), 2.0, 3.0, 3.0])))
+    writer.WriteEvent(
+        self._CreateEventWithDebugNumericSummary(
+            device_name=device_name,
+            op_name='logits/Add',
+            output_slot=0,
+            wall_time=1338,
+            step=8,
+            list_of_values=(list(range(12)) +
+                            [float(tf.int16.as_datatype_enum), 0.0])))
+    writer.Close()
+
+    # Populate the log directory with debugger event for run 'run_foo'.
+    run_foo_directory = os.path.join(self.log_dir, 'run_foo')
+    os.mkdir(run_foo_directory)
+    file_prefix = tf.compat.as_bytes(
+        os.path.join(run_foo_directory, 'events.debugger'))
+    writer = tf.pywrap_tensorflow.EventsWriter(file_prefix)
+    writer.WriteEvent(
+        self._CreateEventWithDebugNumericSummary(
+            device_name=device_name,
+            op_name='layers/Variable',
+            output_slot=0,
+            wall_time=4242,
+            step=42,
+            list_of_values=(list(range(12)) +
+                            [float(tf.int16.as_datatype_enum), 1.0, 8.0])))
+    writer.Close()
+
+    # Start a server that will receive requests and respond with health pills.
+    self.multiplexer = event_multiplexer.EventMultiplexer({
+        '.': self.log_dir,
+        'run_foo': run_foo_directory,
+    })
+    self.debugger_data_server_grpc_port = 4242
+
+    # Fake threading behavior so that threads are synchronous.
+    class FakeThread(object):
```

Comment:
Actually, I spoke too soon. Here we are patching the `start()` method of objects to be created in the future. I'm not sure if there is an easier way to do that.